spi/src/main/java/org/opensearch/jobscheduler/spi/utils/LockService.java

```diff
@@ -244,28 +336,32 @@ private void createLock(final LockModel tempLock, ActionListener<LockModel> list
 
     private void findLock(final String lockId, ActionListener<LockModel> listener) {
         GetRequest getRequest = new GetRequest(LOCK_INDEX_NAME).id(lockId);
-        client.get(getRequest, ActionListener.wrap(
-                response -> {
-                    if (!response.isExists()) {
-                        listener.onResponse(null);
-                    } else {
-                        try {
-                            XContentParser parser = XContentType.JSON.xContent()
-                                    .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE,
-                                            response.getSourceAsString());
-                            parser.nextToken();
-                            listener.onResponse(LockModel.parse(parser, response.getSeqNo(), response.getPrimaryTerm()));
-                        } catch (IOException e) {
-                            logger.error("IOException occurred finding lock", e);
-                            listener.onResponse(null);
+        checkAndUpdateLockMapping(ActionListener.wrap(
```

Comment:
is there really needed here? you are doing a read call, why update the mappings?